File: java/src/com/codechef/beginner/HS08TEST.java
package com.codechef.beginner;

import java.util.*;
import java.lang.*;
import java.io.*;

//Q:- https://www.codechef.com/submit/HS08TEST

public class HS08TEST {

    public static void main (String[] args) throws java.lang.Exception
    {
        Scanner sc = new Scanner(System.in);
        String[] input = sc.nextLine().split(" ");
        double withdraw = Double.parseDouble(input[0]);
        double balance = Double.parseDouble(input[1]);
        double charge = 0.50;
        if(withdraw%5==0 && withdraw<=(balance-charge)){
            balance = balance - withdraw - charge;
        }
        System.out.printf("%.2f %n",balance);
        sc.close();
    }
}

File: python/codechef/begineers/HS08TEST.py
def main() :
	a,b = input().split(" ")
	x = int(a)
	y = float(b)
	z=0.50
	if (x%5==0 and x<=y-z) :
		p = y-z-x
		print("%.2f" % p)
	else:
		print("%.2f" % y)

if __name__ == '__main__':
	main()